components/esp_eth/src/esp_eth_phy_ksz80xx.c

```diff
@@ -397,61 +321,66 @@ static esp_err_t ksz80xx_deinit(esp_eth_phy_t *phy)
     return ret;
 }
 
-esp_eth_phy_t *esp_eth_phy_new_ksz8041(const eth_phy_config_t *config)
+static phy_ksz80xx_t *esp_eth_phy_new_ksz80xx(const eth_phy_config_t *config)
 {
     esp_eth_phy_t *ret = NULL;
     ESP_GOTO_ON_FALSE(config, NULL, err, TAG, "can't set phy config to null");
-    phy_ksz80xx_t *ksz8041 = calloc(1, sizeof(phy_ksz80xx_t));
-    ESP_GOTO_ON_FALSE(ksz8041, NULL, err, TAG, "calloc ksz8041 failed");
-    ksz8041->vendor_model = KSZ8041_MODEL_ID;
-    ksz8041->addr = config->phy_addr;
-    ksz8041->reset_gpio_num = config->reset_gpio_num;
-    ksz8041->reset_timeout_ms = config->reset_timeout_ms;
-    ksz8041->link_status = ETH_LINK_DOWN;
-    ksz8041->autonego_timeout_ms = config->autonego_timeout_ms;
-    ksz8041->parent.reset = ksz80xx_reset;
-    ksz8041->parent.reset_hw = ksz80xx_reset_hw;
-    ksz8041->parent.init = ksz80xx_init;
-    ksz8041->parent.deinit = ksz80xx_deinit;
-    ksz8041->parent.set_mediator = ksz80xx_set_mediator;
-    ksz8041->parent.negotiate = ksz80xx_negotiate;
-    ksz8041->parent.get_link = ksz80xx_get_link;
-    ksz8041->parent.pwrctl = ksz80xx_pwrctl;
-    ksz8041->parent.get_addr = ksz80xx_get_addr;
-    ksz8041->parent.set_addr = ksz80xx_set_addr;
-    ksz8041->parent.advertise_pause_ability = ksz80xx_advertise_pause_ability;
-    ksz8041->parent.loopback = ksz80xx_loopback;
-    ksz8041->parent.del = ksz80xx_del;
-    return &(ksz8041->parent);
+    phy_ksz80xx_t *ksz80xx = calloc(1, sizeof(phy_ksz80xx_t));
+    ESP_GOTO_ON_FALSE(ksz80xx, NULL, err, TAG, "calloc ksz80xx failed");
+    ksz80xx->addr = config->phy_addr;
+    ksz80xx->reset_gpio_num = config->reset_gpio_num;
+    ksz80xx->reset_timeout_ms = config->reset_timeout_ms;
+    ksz80xx->link_status = ETH_LINK_DOWN;
+    ksz80xx->autonego_timeout_ms = config->autonego_timeout_ms;
+    ksz80xx->parent.reset = ksz80xx_reset;
+    ksz80xx->parent.reset_hw = ksz80xx_reset_hw;
+    ksz80xx->parent.init = ksz80xx_init;
+    ksz80xx->parent.deinit = ksz80xx_deinit;
+    ksz80xx->parent.set_mediator = ksz80xx_set_mediator;
+    ksz80xx->parent.negotiate = ksz80xx_negotiate;
+    ksz80xx->parent.get_link = ksz80xx_get_link;
+    ksz80xx->parent.pwrctl = ksz80xx_pwrctl;
+    ksz80xx->parent.get_addr = ksz80xx_get_addr;
+    ksz80xx->parent.set_addr = ksz80xx_set_addr;
+    ksz80xx->parent.advertise_pause_ability = ksz80xx_advertise_pause_ability;
+    ksz80xx->parent.del = ksz80xx_del;
+    return &(ksz80xx->parent);
 err:
     return ret;
 }
 
+esp_eth_phy_t *esp_eth_phy_new_ksz8041(const eth_phy_config_t *config)
+{
+    phy_ksz80xx_t *ksz80xx = esp_eth_phy_new_ksz80xx(config);
+    if (ksz80xx) {
```

Comment:
Consider using `ESP_GOTO_ON_FALSE` instead of `if` statement to follow the ESP-IDF coding style.